src/api/app/models/bs_request_action.rb

```diff
@@ -920,6 +920,8 @@ def expand_targets(ignore_build_state)
   end
 
   def source_access_check!
+    return if Project.is_remote_project?(source_project, true)
```

Comment:
I have asked you exactly that question in the issue and you told me that it's up to the remote instance if it serves this or not....

The `remoteurl` you have pasted is in no way behaving differently in our `PublicController` then without credentials.